java-frontend/src/test/java/org/sonar/java/regex/RegexParserTestUtils.java

```diff
@@ -47,22 +49,32 @@
 
   private RegexParserTestUtils() {}
 
-  public static RegexTree assertSuccessfulParse(String regex) {
+  public static RegexParseResult assertSuccessfulParse(String regex) {
```

Comment:
I'm really not sure about this change. Why not returning always the RegexTree and use the other methods `parseRegex(String regex)` when we need to test the flags.
When I look at all the tests which have been changed to call `getResult()` just after, I feel it reduced the readability of the test.